<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="ns" default=""/>
  <arg name="base_frame" default=""/>
  <arg name="parent_frame" default=""/>
  <arg name="child_frame" default=""/>

  <group>
    <push-ros-namespace namespace="$(var ns)"/>
    <node pkg="extrinsic_tag_based_base_calibrator" exec="extrinsic_tag_based_base_calibrator" name="extrinsic_tag_based_base_calibrator" output="screen">
      <param name="base_frame" value="$(var base_frame)"/>
      <param name="parent_frame" value="$(var parent_frame)"/>
      <param name="child_frame" value="$(var child_frame)"/>

      <!--BA options -->
      <param name="ba_optimize_intrinsics" value="true"/>
      <param name="ba_share_intrinsics" value="true"/>
      <param name="ba_force_shared_ground_plane" value="true"/>
    </node>

    <node pkg="extrinsic_tag_based_base_calibrator" exec="calibrator_ui_node.py" name="calibrator_ui" output="screen"/>
  </group>
</launch>
